app/uk/gov/hmrc/alcoholdutyreturns/controllers/CacheController.scala

```diff
@@ -61,42 +60,45 @@ class CacheController @Inject() (
       }
     }
 
-  def add(): Action[JsValue] =
+  def createUserAnswers(): Action[JsValue] =
     authorise(parse.json).async { implicit request =>
-      withJsonBody[UserAnswers] { userAnswers =>
-        accountService
-          .createUserAnswers(userAnswers)
-          .foldF(
-            err => Future.successful(error(err)),
-            ua =>
-              cacheRepository.add(ua).map { userAnswers =>
-                auditReturnStarted(userAnswers)
-                Ok(Json.toJson(ua))
-              }
-          )
-      }
-    }
+      withJsonBody[ReturnAndUserDetails] { returnAndUserDetails =>
+        val returnId = returnAndUserDetails.returnId
 
-  private def auditReturnStarted(userAnswers: UserAnswers)(implicit hc: HeaderCarrier): Unit = {
-    val maybeRegimeAndObligations = Try {
-      Json.fromJson[RegimeAndObligations](userAnswers.data).asOpt
-    }.toOption.flatten
+        val eitherAccountDetails = for {
```

Comment:
This fixes the issue with having to pull the obligationData back out of the json (and potentially not being able to get it - it was not possible to test this properly. Thus get details from the accountService at this level and put a create method on userAnswers.